Cython/Compiler/Symtab.py

```diff
@@ -1227,14 +1227,12 @@ def __init__(self):
         # which is apparently a special case because it conflicts with C++ bool
         self.declare_var("bool", py_object_type, None, "((PyObject*)&PyBool_Type)")
 
-    def lookup(self, name, language_level=None, str_is_str=None):
-        # 'language_level' and 'str_is_str' are passed by ModuleScope
-        if name == 'str':
-            if str_is_str is None:
-                str_is_str = language_level in (None, 2)
-            if not str_is_str:
-                name = 'unicode'
-        if name == 'long' and language_level == 2:
+    def lookup(self, name, language_level=None):
+        # 'language_level' is passed by ModuleScope
+        if name == 'unicode' or name == 'basestring':
```

Comment:
I don't know how useful it is to map `basestring` to anything. In that if I've got some legacy code with `ininstance(something, basestring)` I'd expect it to match either `str` or `bytes` in Python 3.

I don't think it's worth the effort of making it work, but I also don't know if it's worth keeping around an incorrect mapping that only does half of what it should.

Reply:
`basestring` always mapped to only `str` in Py3. It only included `bytes` in Py2, because that used to be `str` there.

I kept the name purely to avoid breaking existing code that uses it correctly. It might be worth adding a warning about the usage, though.

Reply:
Fair enough - if it's been that way for a while then it probably makes sense to keep it like that.